wandb/sdk/wandb_artifacts.py

```diff
@@ -100,16 +97,20 @@ def _ensure_can_add(self):
         if self._final:
             raise ValueError("Can't add to finalized artifact.")
 
+    @contextlib.contextmanager
```

Comment:
NOTE: new_file now returns a contextmanager (before it returned a file object that could have - but did not have to be - used as a context manager)

I see all the examples used it the method with the "with" statement. but it probably should be more clear in the docs (and eventually this will be typed :)